mantidimaging/gui/windows/spectrum_viewer/presenter.py

```diff
@@ -565,19 +565,75 @@ def setup_fitting_model(self) -> None:
         self.view.scalable_roi_widget.set_parameters(param_names)
         self.view.exportDataTableWidget.set_parameters(param_names)
 
-    def get_init_params_from_roi(self):
+    def get_init_params_from_roi(self) -> None:
         fitting_region = self.view.get_fitting_region()
         init_params = self.model.fitting_engine.get_init_params_from_roi(fitting_region)
         self.view.scalable_roi_widget.set_parameter_values(init_params)
+
+        self.view.fittingDisplayWidget.set_plot_mode("initial")
+
         self.show_initial_fit()
         roi_name = self.view.roiSelectionWidget.current_roi_name
         self.view.exportDataTableWidget.update_roi_data(roi_name=roi_name, params=init_params, status="Initial")
 
-    def show_initial_fit(self):
+    def _plot_initial_fit(self) -> None:
+        assert self.model.tof_data is not None
         init_params = self.view.scalable_roi_widget.get_initial_param_values()
         xvals = self.model.tof_data
         init_fit = self.model.fitting_engine.model.evaluate(xvals, init_params)
-        self.view.fittingDisplayWidget.show_init_fit(xvals, init_fit)
+        self.view.fittingDisplayWidget.show_fit_line(xvals,
+                                                     init_fit,
+                                                     color=(128, 128, 128),
+                                                     label="initial",
+                                                     initial=True)
+
+    def _plot_fitted_curve(self) -> None:
+        assert self.model.tof_data is not None
+        fitted_params = self.view.scalable_roi_widget.get_fitted_param_values()
+        xvals = self.model.tof_data
+        fit = self.model.fitting_engine.model.evaluate(xvals, fitted_params)
+        self.view.fittingDisplayWidget.show_fit_line(xvals, fit, color=(0, 128, 255), label="fit", initial=False)
```

Comment:
This is only called from the tests, so can be removed

Reply:
Thanks for catching this! Have now removed